src/client/executor.rs

```diff
@@ -403,6 +410,13 @@ impl Client {
                     Ok(r) => {
                         self.update_cluster_time(&r, session).await;
                         if r.is_success() {
+                            // Retrieve recovery token from successful response.
+                            if let Some(ref mut session) = session {
```

Comment:
In the event of a deserialization error for an `ok: 1` response, we still want to update the recovery token (e.g. a user provides a `T` to a `find` that doesn't map correctly to their data). The branch for that is a little bit down on lines 463 and 464. We may want some sort of function for this now that it's being done in two places.